src/parquet/file/metadata.h

```diff
@@ -178,6 +178,8 @@ class PARQUET_EXPORT FileMetaData {
   // Return const-pointer to make it clear that this object is not to be copied
   const SchemaDescriptor* schema() const;
 
+  std::unordered_map<std::string, std::string> key_value_metadata() const;
```

Comment:
I'm a bit torn about exposing `unordered_map<string, string>` in the public API. The alternative would be something like:

```c++
class KeyValueMetadata {
 public:
  int length() const;

  const std::string& key(int i) const;
  const std::string& value(int i) const;

  void Append(const std::string& key, const std::string& value) const;

 private:
  std::vector<std::string> keys_;
  std::vector<std::string> values_;
};
```

then

```c++
const KeyValueMetadata& key_value_metadata() const;
```